jgrapht-core/src/main/java/org/jgrapht/alg/scoring/AlphaCentrality.java

```diff
@@ -0,0 +1,281 @@
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, double exogenousFactor)
+    {
+        this(g, dampingFactor, exogenousFactor, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactormap the exogenous factor map
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, Map<V, Double> exogenousFactormap)
+    {
+        this(g, dampingFactor, exogenousFactormap, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
+     * @param maxIterations the maximum number of iterations to perform
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, double exogenousFactor, int maxIterations)
+    {
+        this(g, dampingFactor, exogenousFactor, maxIterations, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactormap the exogenous factor map
+     * @param maxIterations the maximum number of iterations to perform
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, Map<V, Double> exogenousFactormap, int maxIterations)
+    {
+        this(g, dampingFactor, exogenousFactormap, maxIterations, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
+     * @param maxIterations the maximum number of iterations to perform
+     * @param tolerance the calculation will stop if the difference of AlphaCentrality values between
+     *        iterations change less than this value
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, double exogenousFactor, int maxIterations, double tolerance)
+    {
+        this.g = g;
+        this.scores = new HashMap<>();
+
+        if (maxIterations <= 0) {
+            throw new IllegalArgumentException("Maximum iterations must be positive");
+        }
+
+        if (dampingFactor < 0.0 || dampingFactor > 1.0) {
+            throw new IllegalArgumentException("Damping factor not valid");
+        }
+
+        if (tolerance <= 0.0) {
+            throw new IllegalArgumentException("Tolerance not valid, must be positive");
+        }
+
+        Map<V, Double> exofactor = new HashMap<>();
```

Comment:
Rather than actually building the map, you could change the run method to take a function as input.

Reply:
Can you please explain me a bit more..? I didn't get you exactly..

Reply:
Run parameter type would be Function<Vertex, Double>.

For scalar, apply always returns the constant value.

For map, apply does the map lookup.

If this still isn't clear, you'll want to read up on Java 8 functions/lambda.

Reply:
What I think @jsichi means is you can convert exofactor from a `Map<V, Double>` to a `ToDoubleFunction<V>`.

So instead of:

```java
Map<V, Double> exofactor = ...
run(dampingFactor, exofactor, maxIterations, tolerance);

. . .

private void run(double dampingFactor, Map<V, Double> exofactormap, int maxIterations, double tolerance)
{
    ...
    double vNewValue = contribution + exofactormap.get(v);
    ...
}
```

perhaps something like:

```java
ToDoubleFunction<V> exofactorFunction = v -> exogenousFactor;
run(dampingFactor, exofactorFunction, maxIterations, tolerance);

. . .

private void run(double dampingFactor, ToDoubleFunction<V> exofactorFunction, int maxIterations, double tolerance)
{
    ...
    double vNewValue = contribution + exofactorFunction.applyAsDouble(v);
    ...
}
```

The advantage of this is it would allow the exofactors of each vertex to be computed only when they're really needed, which should save space and time compared to creating a map upfront.

Reply:
Well, I say "computing", but in this case the code is just retrieving an effectively constant exogenous factor, so no computation is involved per se, but it would still save space and time that it would spend otherwise creating a map. :)

Reply:
Right, what he said.  :)

I spend 90% of my time in scala land these days, so when I mention anything from Java 8 I am waving my hands.

Reply:
Done @jsichi. Moreover should I squash these 2 new commits or it is fine?